docs/imagecustomizer/api/storage.md

```diff
@@ -0,0 +1,367 @@
+  - [disks \[disk\[\]\]](#disks-disk)
+    - [disk type](#disk-type)
+      - [partitionTableType \[string\]](#partitiontabletype-string)
+      - [maxSize \[uint64\]](#maxsize-uint64)
+      - [partitions \[partition\[\]\]](#partitions-partition)
+        - [partition type](#partition-type)
+          - [id \[string\]](#id-string)
+          - [label \[string\]](#label-string)
+          - [start \[uint64\]](#start-uint64)
+          - [end \[uint64\]](#end-uint64)
+          - [size \[uint64\]](#size-uint64)
+          - [type \[string\]](#type-string)
+      - [filesystems \[filesystem\[\]\]](#filesystems-filesystem)
+        - [filesystem type](#filesystem-type)
+          - [deviceId \[string\]](#deviceid-string)
+          - [type \[string\]](#type-string-1)
+          - [mountPoint \[mountPoint\]](#mountpoint-mountpoint)
+          - [mountPoint type](#mountpoint-type)
+  - [verity \[verity\[\]\]](#verity-verity)
+    - [verity type](#verity-type)
+      - [id \[string\]](#id-string-1)
+      - [name \[string\]](#name-string)
+      - [dataDeviceId \[string\]](#datadeviceid-string)
+      - [hashDeviceId \[string\]](#hashdeviceid-string)
+      - [corruptionOption \[string\]](#corruptionoption-string)
+
+## bootType [string]
+
+Specifies the boot system that the image supports.
+
+Supported options:
+
+- `legacy`: Support booting from BIOS firmware.
+
+  When this option is specified, the partition layout must contain a partition with the
+  `bios-grub` flag.
+
+- `efi`: Support booting from UEFI firmware.
+
+  When this option is specified, the partition layout must contain a partition with the
+  `esp` flag.
+
+## resetPartitionsUuidsType [string]
+
+Specifies that the partition UUIDs and filesystem UUIDs should be reset.
+
+Value is optional.
+
+This value cannot be specified if [storage](#storage) is specified (since
+customizing the partition layout resets all the UUIDs anyway).
+
+If this value is specified, then [os.bootLoader.resetType](./os.md#resettype-string)
+must also be specified.
+
+Supported options:
+
+- `reset-all`: Resets the partition UUIDs and filesystem UUIDs for all the partitions.
+
+Example:
+
+```yaml
+storage:
+  resetPartitionsUuidsType: reset-all
+
+os:
+  bootloader:
+    resetType: hard-reset
+```
+
+## disks [[disk](#disk-type)[]]
+
+Contains the options for provisioning disks and their partitions.
+
+### disk type
+
+Specifies the properties of a disk, including its partitions.
+
+#### partitionTableType [string]
+
+Specifies how the partition tables are laid out.
+
+Supported options:
+
+- `gpt`: Use the GUID Partition Table (GPT) format.
+
+#### maxSize [uint64]
+
+The size of the disk.
+
+Supported format: `<NUM>(K|M|G|T)`: A size in KiB (`K`), MiB (`M`), GiB (`G`),
+or TiB
+(`T`).
+
+Must be a multiple of 1 MiB.
+
+#### partitions [[partition](#partition-type)[]]
+
+The partitions to provision on the disk.
+
+##### partition type
+
+<div id="partition-id"></div>
+
+###### id [string]
```

Comment:
It is a bad idea to have headers that are nested this deep. There usually aren't enough ways to distinguish between the headers at this point.